### PR TITLE
[subset] Add amalgam harfbuzz-subset.cc

### PR DESCRIPTION
**Pull request by khaledhosny**
Fixes https://github.com/harfbuzz/harfbuzz/issues/3733

### CODE REVIEW

File: src/Makefile.am [hunk 1]
```diff
@@ -160,6 +160,23 @@ pkginclude_HEADERS += $(HB_SUBSET_headers)
...
+	$(AM_V_GEN) \
+	for f in \
+		$(HB_BASE_sources) \
+		$(HB_GLIB_sources) \
+		$(HB_FT_sources) \
+		$(HB_GRAPHITE2_sources) \
+		$(HB_UNISCRIBE_sources) \
+		$(HB_GDI_sources) \
+		$(HB_DIRECTWRITE_sources) \
+		$(HB_CORETEXT_sources) \
```
Comment (behdad) [inline src/Makefile.am:173]: You *just* need `HB_SUBSET_sources` here to build harfbuzz-subset library.

**Comment by behdad** [comment]
Maybe add a sort command to the autotools path as well?

File: src/Makefile.am (hunk 1 above)
Comment (khaledhosny) [reply src/Makefile.am:173]: The idea is to build a (static) standalone hb-subset, but may be that is too WASM-specific. I guess one can always `c++ harfbuzz.cc harfbuzz-subset.cc` to achieve the same.

Comment (behdad) [reply src/Makefile.am:173]: > I guess one can always c++ harfbuzz.cc harfbuzz-subset.cc to achieve the same.

Yes, if that works, I prefer that. Then `harfbuzz-subset.cc` will be equivalent to `libharfbuzz-subset`.

Comment (behdad) [reply src/Makefile.am:173]: Okay, then at least just include the base HB sources. No need for any auxiliary sources; unless you need those as well.

Comment (khaledhosny) [reply src/Makefile.am]: Done.

**Comment by khaledhosny** [comment]
> Maybe add a sort command to the autotools path as well?

Done.

**Pull request merged**